gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -40,36 +65,126 @@
+      producer = DagManagerUtils.getSpecProducer(dagNode);
+      TimingEvent jobOrchestrationTimer = eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.JOB_ORCHESTRATED);
+
+      // Increment job count before submitting the job onto the spec producer, in case that throws an exception.
+      // By this point the quota is allocated, so it's imperative to increment as missing would introduce the potential to decrement below zero upon quota release.
+      // Quota release is guaranteed, despite failure, because exception handling within would mark the job FAILED.
+      // When the ensuing kafka message spurs DagManager processing, the quota is released and the counts decremented
+      // Ensure that we do not double increment for flows that are retried
+      if (dagNode.getValue().getCurrentAttempts() == 1) {
+        DagManagementTaskStreamImpl.getDagManagerMetrics().incrementRunningJobMetrics(dagNode);
+      }
+      // Submit the job to the SpecProducer, which in turn performs the actual job submission to the SpecExecutor instance.
+      // The SpecProducer implementations submit the job to the underlying executor and return when the submission is complete,
+      // either successfully or unsuccessfully. To catch any exceptions in the job submission, the DagManagerThread
+      // blocks (by calling Future#get()) until the submission is completed.
+      Future<?> addSpecFuture = producer.addSpec(jobSpec);
+      dagNode.getValue().setJobFuture(com.google.common.base.Optional.of(addSpecFuture));
+
+      addSpecFuture.get();
+
+      jobMetadata.put(TimingEvent.METADATA_MESSAGE, producer.getExecutionLink(addSpecFuture, specExecutorUri));
+      // Add serialized job properties as part of the orchestrated job event metadata
+      jobMetadata.put(JobExecutionPlan.JOB_PROPS_KEY, dagNode.getValue().toString());
+      jobOrchestrationTimer.stop(jobMetadata);
+      log.info("Orchestrated job: {} on Executor: {}", DagManagerUtils.getFullyQualifiedJobName(dagNode), specExecutorUri);
+      DagManagementTaskStreamImpl.getDagManagerMetrics().incrementJobsSentToExecutor(dagNode);
+    } catch (Exception e) {
+      TimingEvent jobFailedTimer = eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.JOB_FAILED);
+      String message = "Cannot submit job " + DagManagerUtils.getFullyQualifiedJobName(dagNode) + " on executor " + specExecutorUri;
+      log.error(message, e);
+      jobMetadata.put(TimingEvent.METADATA_MESSAGE, message + " due to " + e.getMessage());
+      if (jobFailedTimer != null) {
+        jobFailedTimer.stop(jobMetadata);
+      }
+    }
   }
 
   @Override
-  protected void commit(DagManagementStateStore dagManagementStateStore, Optional<Dag<JobExecutionPlan>> dag) {
+  protected void sendNotification(Optional<Dag<JobExecutionPlan>> result, EventSubmitter eventSubmitter) {
     throw new UnsupportedOperationException("Not yet implemented");
```

Comment:
we don't really want to throw this, so better would be empty impl.

(BTW, a `LaunchDagProcTest` almost certainly would have caught this)